### PR TITLE
MULE-15119: Default scheduler-pools.conf for 1 core and 256 MB heap

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
         <groupId>org.mule.services</groupId>
         <artifactId>mule-core-services-parent</artifactId>
         <version>1.2.0-SNAPSHOT</version>
+        <relativePath></relativePath>
```
Comment (marianogonzalez) [inline pom.xml]: ???

File: src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java [hunk 2]
```diff
@@ -63,8 +64,16 @@
...
   private static final String NUMBER_OR_VAR_REGEXP = "([0-9]+(\\.[0-9]+)?)|cores|mem";
+  private static final String FORMULA_FUNCTION_PARAM =
+      "(" + NUMBER_OR_VAR_REGEXP + ")?(\\s*[-+\\/*\\(\\)]\\s*(" + NUMBER_OR_VAR_REGEXP + ")?)*";
   private static final Pattern POOLSIZE_PATTERN =
-      compile("(" + NUMBER_OR_VAR_REGEXP + ")?(\\s*[-+\\/*\\(\\)]\\s*(" + NUMBER_OR_VAR_REGEXP + ")?)*");
+      compile("^" + FORMULA_FUNCTION_PARAM
+          + "|max\\s*\\(\\s*(" + FORMULA_FUNCTION_PARAM + ")?\\s*,\\s*(" + FORMULA_FUNCTION_PARAM + ")?\\s*\\)"
+          + "|min\\s*\\(\\s*(" + FORMULA_FUNCTION_PARAM + ")?\\s*,\\s*(" + FORMULA_FUNCTION_PARAM + ")?\\s*\\)$");
+
+  public static void main(String[] args) {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: gotcha

File: src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java [hunk 3]
```diff
@@ -176,10 +185,13 @@ private static OptionalInt resolveExpression(Properties properties, String propN
       return OptionalInt.empty();
     }
 
-    final String property = properties.getProperty(propName).trim().toLowerCase();
+    String property = properties.getProperty(propName).trim().toLowerCase();
```
Comment (marianogonzalez) [inline src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: superflous change

File: src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java [hunk 4]
```diff
@@ -176,10 +185,13 @@ private static OptionalInt resolveExpression(Properties properties, String propN
...
     if (!POOLSIZE_PATTERN.matcher(property).matches()) {
       throw new DefaultMuleException(propName + ": Expression not valid");
     }
+
+    property = property.replaceAll("(max|min)", "Math.$1");
```
Comment (marianogonzalez) [inline src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: static import... oh wait..

File: src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java [hunk 5]
```diff
@@ -211,7 +223,7 @@ private static void validateNumber(String propName, long result, boolean allowZe
   private int cpuLightPoolSize = 2 * cores;
   private int ioQueueSize = 0;
   private int ioCorePoolSize = cores;
-  private int ioMaxPoolSize = (int) (cores + ((mem - 245760) / 5120));
+  private int ioMaxPoolSize = (int) max(2, cores + ((mem - 245760) / 5120));
```
Comment (marianogonzalez) [inline src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: isn't this laying to the user? if the user specifies `1`.. won't this override that and change it to `2`? That's not necessarily a bad thing, but a log message should be printed if that's the case

File: pom.xml (hunk 1 above)
Comment (elrodro83) [reply pom.xml]: piggyback. avoids a warning thrown by maven (https://stackoverflow.com/questions/6003831/parent-relativepath-points-at-my-com-mycompanymyproject-instead-of-org-apache)

File: src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java (hunk 5 above)
Comment (elrodro83) [reply src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: nop, this is the default for when the property does not exist in the config file

File: src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java (hunk 3 above)
Comment (elrodro83) [reply src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: no, i need to write this var below

**Pull request merged**